src/Microsoft.AspNet.Mvc.Core/Infrastructure/MvcRouteHandler.cs

```diff
@@ -57,7 +58,7 @@ public VirtualPathData GetVirtualPath(VirtualPathContext context)
             var actionDescriptor = await _actionSelector.SelectAsync(context);
             if (actionDescriptor == null)
             {
-                _logger.LogVerbose("No actions matched the current request.");
+                _logger.NoMatchingActions(context.HttpContext);
```

Comment:
I've actually made a bunch of conflicting changes in this file, you can revert all of these.